go-controller/pkg/kubevirt/dhcp_test.go

```diff
@@ -46,101 +31,175 @@ var _ = Describe("Kubevirt", func() {
+			expectedDHCPConfigs: dhcpConfigs{
+				V6: &nbdb.DHCPOptions{
+					Cidr: "2002:0:0:1234::/64",
+					ExternalIDs: map[string]string{
+						"k8s.ovn.org/owner-controller": "defaultController",
+						"k8s.ovn.org/owner-type":       "VirtualMachine",
+						"k8s.ovn.org/name":             "namespace1/foo1",
+						"k8s.ovn.org/cidr":             "2002.0.0.1234../64",
+						"k8s.ovn.org/id":               "defaultController:VirtualMachine:namespace1/foo1:2002.0.0.1234../64",
+						"k8s.ovn.org/zone":             "local",
+					},
+					Options: map[string]string{
+						"server_id":  "0a:58:6d:6d:c1:50",
+						"fqdn":       `"foo1"`,
+						"dns_server": "2001:1:2:3:4:5:6:7",
+					},
+				},
+			},
 		}),
-		Entry("Dual stack and k8s dns", dhcpTest{
+		Entry("Dual stack and dns", dhcpTest{
 			cidrs:          []string{"192.168.25.0/24", "2002:0:0:1234::/64"},
 			controllerName: "defaultController",
 			namespace:      "namespace1",
 			vmName:         "foo1",
-			dns:            svc("kube-system", "kube-dns", []string{"192.167.23.44", "2001:1:2:3:4:5:6:7"}),
+			opts: []DHCPConfigsOpt{
+				WithIPv4DNServer("192.167.23.44"),
+				WithIPv6DNServer("2001:1:2:3:4:5:6:7"),
+			},
 			expectedDHCPConfigs: dhcpConfigs{
-				V4: ComposeDHCPv4Options("192.168.25.0/24", "192.167.23.44", "defaultController", key("namespace1", "foo1")),
-				V6: ComposeDHCPv6Options("2002:0:0:1234::/64", "2001:1:2:3:4:5:6:7", "defaultController", key("namespace1", "foo1")),
+				V4: &nbdb.DHCPOptions{
+					Cidr: "192.168.25.0/24",
+					ExternalIDs: map[string]string{
+						"k8s.ovn.org/owner-controller": "defaultController",
+						"k8s.ovn.org/owner-type":       "VirtualMachine",
+						"k8s.ovn.org/name":             "namespace1/foo1",
+						"k8s.ovn.org/cidr":             "192.168.25.0/24",
+						"k8s.ovn.org/id":               "defaultController:VirtualMachine:namespace1/foo1:192.168.25.0/24",
+						"k8s.ovn.org/zone":             "local",
+					},
+					Options: map[string]string{
+						"lease_time": "3500",
+						"server_id":  ARPProxyIPv4,
+						"server_mac": ARPProxyMAC,
+						"hostname":   `"foo1"`,
+						"dns_server": "192.167.23.44",
+					},
+				},
+				V6: &nbdb.DHCPOptions{
+					Cidr: "2002:0:0:1234::/64",
+					ExternalIDs: map[string]string{
+						"k8s.ovn.org/owner-controller": "defaultController",
+						"k8s.ovn.org/owner-type":       "VirtualMachine",
+						"k8s.ovn.org/name":             "namespace1/foo1",
+						"k8s.ovn.org/cidr":             "2002.0.0.1234../64",
+						"k8s.ovn.org/id":               "defaultController:VirtualMachine:namespace1/foo1:2002.0.0.1234../64",
+						"k8s.ovn.org/zone":             "local",
+					},
+					Options: map[string]string{
+						"server_id":  "0a:58:6d:6d:c1:50",
+						"fqdn":       `"foo1"`,
+						"dns_server": "2001:1:2:3:4:5:6:7",
+					},
+				},
 			},
 		}),
-		Entry("Dual stack and k8s dns with ipv4 only", dhcpTest{
+
+		Entry("Dual stack and dns with ipv4 only", dhcpTest{
```

Comment:
Is this test about a dual-stack cluster with only IPv4 DNS ?

Reply:
Yes, renaming the test to  `Dual stack with single IPv4 dns server`